### PR TITLE
Add org.kde.ktorrent (KTorrent)

### PR DESCRIPTION
**Pull request by Siosm**
Upstream appdata.patch: https://invent.kde.org/network/ktorrent/-/merge_requests/17

### CODE REVIEW

**Comment by Siosm** [comment]
bot, build org.kde.ktorrent

**Comment by cho2** [comment]
```
Warning: Audio notification requested, but sound file from notifyrc file was not found, aborting audio notification
```
Settings > Configure Notifications, there is no audio file for notifications.

File: org.kde.ktorrent.json [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        "/lib/cmake",
+        "/lib/pkgconfig",
+        "/share/man"
+    ],
+    "modules": [
+        {
+            "name": "boost",
+            "buildsystem": "simple",
+            "build-commands": [
+                "./bootstrap.sh --prefix=${FLATPAK_DEST}",
```
Comment (eszlari) [inline org.kde.ktorrent.json:26]: Is bootstrap necessary, if you just copy the headers?

File: org.kde.ktorrent.json [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.kde.org/stable/qca/2.3.1/qca-2.3.1.tar.xz",
+                    "sha256": "c13851109abefc4623370989fae3a745bf6b1acb3c2a13a8958539823e974e4b"
+                }
+            ]
+        },
+        {
+            "name": "libktorrent",
```
Comment (eszlari) [inline org.kde.ktorrent.json:57]: `-DBUILD_TESTING=OFF`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# KTorrent (org.kde.ktorrent)
+
+[Homepage](https://kde.org/applications/ktorrent) |
+[User documentation](https://docs.kde.org/stable5/en/extragear-network/ktorrent/index.html) |
+[Sources](https://invent.kde.org/network/ktorrent)
+
+## Reporting bugs
```
Comment (eszlari) [inline README.md:7]: Github issue template?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  application, please file an [Issue here][issue].
+
+- If the bug is also reproducible with the **non-Flatpak version** of this
+  application, please take a look at the [currently opened bugs][bugs] and if
+  it has not been reported yet, file a bug at
+  [bugs.kde.org](https://bugs.kde.org).
+
+**If you have any doubt, please file an [Issue here][issue].**
+
+## Permissions rationale
```
Comment (eszlari) [inline README.md:22]: I would only document the non-obvious ones (`--filesystem=host`) inside the manifest and drop the rest.

Comment (nedrichards) [reply README.md:22]: Yeah, you only need to document anything outside the unrestricted permissions - and even then only *really* in this review. https://github.com/flathub/flathub/wiki/App-Requirements#unrestricted-permissions

File: org.kde.ktorrent.json [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                    "url": "https://dl.bintray.com/boostorg/release/1.74.0/source/boost_1_74_0.tar.bz2",
+                    "sha256": "83bfc1507731a0906e387fc28b7ef5417d591429e51e788417fe9ff025e116b1"
+                }
+            ]
+        },
+        {
+            "name": "qca",
+            "buildsystem": "cmake-ninja",
+            "config-opts": [
+                "-DQT4_BUILD=OFF",
```
Comment (tsdgeos) [inline org.kde.ktorrent.json:43]: you don't need this, there's no qt4 build anymore, you don't need to set the QCA_SUFFIx either

**Comment by Siosm** [comment]
> ```
> Warning: Audio notification requested, but sound file from notifyrc file was not found, aborting audio notification
> ```
> 
> Settings > Configure Notifications, there is no audio file for notifications.

I would have thought that this would be part of the platform but I might have missed something here. Should I also add sound access?

File: org.kde.ktorrent.json (hunk 1 above)
Comment (Siosm) [reply org.kde.ktorrent.json]: Fixed, thanks!

File: org.kde.ktorrent.json (hunk 2 above)
Comment (Siosm) [reply org.kde.ktorrent.json:52]: Fixed, thanks!

File: README.md (hunk 3 above)
Comment (Siosm) [reply README.md:7]: That's a good idea. Maybe we should do that for all the KDE repos?

Comment (Siosm) [reply README.md:7]: https://invent.kde.org/teams/flathub/issues/-/issues/4#note_119906

File: README.md (hunk 4 above)
Comment (Siosm) [reply README.md:22]: I've kept only the restricted ones. I would prefer that we keep those explanations in the README to make them easier to discover for potential contributors or users looking at the repo (and the json file format does not really support comments). Maybe we could switch to in-manifest comments once we move to yaml.

File: org.kde.ktorrent.json (hunk 5 above)
Comment (Siosm) [reply org.kde.ktorrent.json]: Fixed, thanks!

**Comment by Siosm** [comment]
bot, build org.kde.ktorrent

**Comment by Siosm** [comment]
For a yet unknown reason, I also get a dialog with the following error on each startup:
```
Unknown host geolite.maxmind.com: Host not found
```

**Comment by tsdgeos** [comment]
> For a yet unknown reason, I also get a dialog with the following error on each startup:
> 
> ```
> Unknown host geolite.maxmind.com: Host not found
> ```

Nothing unknown about it, ktorrent code queries that domain and that domain doesn't exist anymore

**Comment by tsdgeos** [comment]
This is bad i think
"Warning: env says KDE is running but SNI unavailable -- check KDE_FULL_SESSION and XDG_CURRENT_DESKTOP"
this causes ktorrent to "minmize"  itself to the notification area but doesn't really work because can't talk to it.

Are you running Plasma so you can try to debug what's wrong? I'm guessing we need to punch some dbus hole somewhere

**Comment by Siosm** [comment]
bot, build org.kde.ktorrent

File: README.md (hunk 4 above)
Comment (tsdgeos) [reply README.md:22]: "Unrestricted permissions

The following permissions can be freely used:"

Unrestricted by what? who? is this a flatpak policy or somehow flathub has created their own policy on top of permissions? I read those two sentences and i get no understanding of what they really mean. 

Can someone please clarify?

**Comment by Siosm** [comment]
bot, build org.kde.ktorrent

**Comment by Siosm** [comment]
We have several options to work around / fix the geolite host unknown issue:

  - Port KTorrent to the new [libmaxminddb](https://github.com/maxmind/libmaxminddb) library.
  - Patch the sources to download the database from a [new unofficial legacy provider](https://mailfud.org/geoip-legacy/) ([from Arch Linux package](https://github.com/archlinux/svntogit-packages/blob/packages/geoip-database/trunk/PKGBUILD)).
  - Use the database from the same unofficial source in point 2 but get it and install it from the [versioned and Arch Linux hosted source server](https://sources.archlinux.org/other/packages/geoip-database/)
  - Something else?

I pushed an initial change for point 3 while we consider how to do point 1.

**Comment by Siosm** [comment]
bot, build org.kde.ktorrent

**Comment by nedrichards** [comment]
For now I think 3 is good but we should likely try and fix with 1 upstream.

**Comment by tsdgeos** [comment]
Seems reasonable to me. I'd say good enough to release

**Pull request closed without merging**